From ae7726abe41887192ee2fd47aeb4abf8c139ff75 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 31 Jan 2020 14:53:09 +0200
Subject: [PATCH] [SW-7984] goya2: weaken preboot_version()

Since in goya2 there is no BTL so version should look different
Once goya2 code is added proper method

Change-Id: I1805a83737a8ad1d9bab289398413a7266a684e1
---

diff --git a/app/preboot.c b/app/preboot.c
index 2616853..a59076e 100644
--- a/app/preboot.c
+++ b/app/preboot.c
@@ -97,7 +97,7 @@
 static const char *preboot_ver_str = PREBOOT_VER_STR;
 static char preboot_btl_ver_str[VERSION_MAX_LEN] = "BTL version ";
 
-static void preboot_version(void)
+__attribute__((weak)) void preboot_version(void)
 {
 	unsigned long offset = (unsigned long) preboot_btl_ver_str;
 	u32 btlver = regmap_readl(mmPSOC_GLOBAL_CONF_SCRATCHPAD_28);
@@ -132,7 +132,7 @@
 			mmPSOC_GLOBAL_CONF_SCRATCHPAD_28);
 }
 #else	/* TARGET_IS_TESTER */
-static inline void preboot_version(void) {}
+__attribute__((weak)) void preboot_version(void) {}
 #endif	/* TARGET_IS_TESTER */
 
 void preboot_main(void)
